packages/@expo/metro-config/src/serializer/serializeChunks.ts

```diff
@@ -331,13 +339,13 @@ class Chunk {
       includeBytecode,
     }: { includeSourceMaps?: boolean; includeBytecode?: boolean }
   ): Promise<SerialAsset[]> {
-    const jsCode = this.serializeToCode(serializerConfig, chunks);
+    // Create hash without wrapping to prevent it changing when the wrapping changes.
+    const outputFile = this.getFilenameForConfig(serializerConfig);
+    // We already use a stable hash for the output filename, so we'll reuse that for the debugId.
+    const debugId = stringToUUID(path.basename(outputFile, path.extname(outputFile)));
+    const jsCode = this.serializeToCode(serializerConfig, { chunks, debugId });
```

Comment:
To confirm I understand this correctly the file name includes a hash based on the output of `serializeToCodeWithTemplates` this works and changes with source change.

Reply:
correct, there's a single hash that we're quickly pulling out here. In a future version we can clean this all up.